### PR TITLE
UCP/AM: AM rndv fixes

### PR DESCRIPTION
**Pull request by brminich**
## What
Several UCP AM improvements/fixes:
- Adjust max AM header based on RTS size
- Fix small `UCX_RNDV_THRESH` (smaller than `max_short`) for AM rndv
- A couple of fixes in `ucp_am_send_req`
- Tests for the above

### CODE REVIEW

**Comment by brminich** [comment]
#5937 

bot:pipe:retest

**Comment by brminich** [comment]
issue is #5937

bot:pipe:retest

**Comment by brminich** [comment]
@dmitrygx, @hoopoepg, please review

File: src/ucp/core/ucp_ep.c [hunk 1]
```diff
@@ -1331,7 +1331,8 @@ static void ucp_ep_config_set_rndv_thresh(ucp_worker_t *worker,
                                           ucp_lane_index_t *lanes,
                                           size_t min_rndv_thresh,
                                           size_t max_rndv_thresh,
-                                          ucp_rndv_thresh_t *thresh)
+                                          ucp_rndv_thresh_t *thresh,
```
Comment (dmitrygx) [inline src/ucp/core/ucp_ep.c:1334]: minor: should we add _p suffix for `thresh` and `max_short_to_adjust` arguments?

File: src/ucp/core/ucp_ep.h [hunk 2]
```diff
@@ -334,6 +334,9 @@ struct ucp_ep_config {
         /* Protocols used for am operations */
         const ucp_request_send_proto_t   *proto;
         const ucp_request_send_proto_t   *reply_proto;
+
+        /* Maximal size for eager short. */
```
Comment (dmitrygx) [inline src/ucp/core/ucp_ep.h:338]: no need for trailing `.`

**Review by dmitrygx — CHANGES_REQUESTED** [review]

File: src/ucp/core/ucp_ep.c (hunk 1 above)
Comment (brminich) [reply src/ucp/core/ucp_ep.c:1334]: we do not have common rule for that. Is it to mark `out` parameters or relevant for `worker`, `config` and `lanes` as well?

File: src/ucp/core/ucp_ep.h (hunk 2 above)
Comment (brminich) [reply src/ucp/core/ucp_ep.h:338]: ok, will fix with other comments

**Comment by brminich** [comment]
@yosefe, please review

**Review by yosefe — APPROVED** [review]

File: src/ucp/core/ucp_ep.h (hunk 2 above)
Comment (brminich) [reply src/ucp/core/ucp_ep.h]: fixed

**Comment by brminich** [comment]
@dmitrygx, do you have other comments?

**Review by dmitrygx — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**